objects/src/accounts/account_id.rs

```diff
@@ -134,336 +129,471 @@ impl FromStr for AccountStorageMode {
     }
 }
 
+// ACCOUNT VERSION
+// ================================================================================================
+
+/// The version of an [`AccountId`].
+///
+/// Each version has a public associated constant, e.g. [`AccountVersion::VERSION_0`].
+#[derive(Debug, Clone, PartialEq, Eq, Copy)]
+pub struct AccountVersion(u8);
+
+impl AccountVersion {
+    // CONSTANTS
+    // --------------------------------------------------------------------------------------------
+
+    const VERSION_0_NUMBER: u8 = 0;
+    /// Version 0 of the [`AccountId`].
+    pub const VERSION_0: AccountVersion = AccountVersion(Self::VERSION_0_NUMBER);
+
+    // PUBLIC ACCESSORS
+    // --------------------------------------------------------------------------------------------
+
+    /// Returns the version number.
+    pub const fn version_num(&self) -> u8 {
+        self.0
+    }
+}
+
 // ACCOUNT ID
 // ================================================================================================
 
-/// Unique identifier of an account.
+/// The identifier of an [`Account`](crate::accounts::Account).
+///
+/// # Layout
 ///
-/// Account ID consists of 1 field element (~64 bits). The most significant bits in the id are used
-/// to encode the account' storage and type.
+/// An `AccountId` consists of two field elements and is layed out as follows:
 ///
-/// The top two bits are used to encode the storage type. The values [PRIVATE] and [PUBLIC]
-/// encode the account's storage type. The next two bits encode the account type. The values
-/// [FUNGIBLE_FAUCET], [NON_FUNGIBLE_FAUCET], [REGULAR_ACCOUNT_IMMUTABLE_CODE], and
-/// [REGULAR_ACCOUNT_UPDATABLE_CODE] encode the account's type.
+/// ```text
+/// 1st felt: [zero bit | random (55 bits) | storage mode (2 bits) | type (2 bits) | version (4 bits)]
+/// 2nd felt: [anchor_epoch (16 bits) | random (40 bits) | 8 zero bits]
+/// ```
+///
+/// # Generation
+///
+/// An `AccountId` is a commitment to a user-generated seed, the code and storage of an account and
+/// to a certain hash of an epoch block of the blockchain. An id is generated by picking an epoch
```

Comment:
nit: At some points (in this doc comment and other files; even in the doc articles IIRC) `ID` is sometimes written as `id`. It would be great to choose one style and use it consistently throughout all projects. I personally prefer `ID`, but don't mind either way.

Reply:
Thanks for pointing this out, I added this to the follow-up tasks as well as it is of lower priority.